pkg/core/storelimit/sliding_window.go

```diff
@@ -30,30 +35,53 @@ var _ StoreLimit = &SlidingWindows{}
 type SlidingWindows struct {
 	mu      syncutil.RWMutex
 	windows []*window
+	lastSum float64
 }
 
 // NewSlidingWindows is the construct of SlidingWindows.
-func NewSlidingWindows(cap float64) *SlidingWindows {
-	if cap < 0 {
-		cap = minSnapSize
-	}
+func NewSlidingWindows() *SlidingWindows {
 	windows := make([]*window, constant.PriorityLevelLen)
 	for i := 0; i < int(constant.PriorityLevelLen); i++ {
-		windows[i] = newWindow(int64(cap) >> i)
+		windows[i] = newWindow(int64(defaultWindowSize) >> i)
 	}
 	return &SlidingWindows{
 		windows: windows,
 	}
 }
 
-// Reset resets the capacity of the sliding windows.
-// It doesn't clear all the used, only set the capacity.
-func (s *SlidingWindows) Reset(cap float64, typ Type) {
+// Version returns v2
+func (s *SlidingWindows) Version() string {
+	return VersionV2
+}
+
+// Feedback is used to update the capacity of the sliding windows.
+func (s *SlidingWindows) Feedback(e float64) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	// If the limiter is available, we don't need to update the capacity.
+	if s.windows[constant.Low].available() {
+		return
+	}
+	s.lastSum += e
+	// There are two constants to control the proportion of the sum and the current error.
+	// The sum of the error is used to ensure the capacity is more stable even if the error is zero.
+	// In the final scene, the sum of the error should be stable and the current error should be zero.
+	cap := defaultProportion*e + defaultIntegral*s.lastSum
```

Comment:
If executing_duration is always greater than waiting_duration, will the cap will go up infinitely?

Reply:
no, the cap will only go up iff the space is not available.